x-pack/platform/plugins/shared/fleet/server/routes/agent_policy/handlers.ts

```diff
@@ -540,6 +540,13 @@ export const updateAgentPolicyHandler: FleetRequestHandler<
   logger.debug(`updating policy [${request.params.agentPolicyId}] in space [${spaceId}]`);
 
   try {
+    const authzFleetAgentsAll = fleetContext.authz.fleet.allAgents;
+
+    if (data.required_versions && !authzFleetAgentsAll) {
```

Comment:
Should we throw if user provide the same `required_version` ? I think our UI is most of the time sending the whole policy

Reply:
Yeah I agree, I was holding on that on trying to decide where to place that check. I wanted to keep this kind of auth check in the handler, but also didn't want to redundantly pull the existing policy with a get, when we do that later in the `agentPolicyService.update`. I think what I'll do here is pass an optional opt to `agentPolicyService.update` that is the result of the auth check, and incorporate it into the `validateRequiredVersions` check.